quarkus/config-api/src/main/java/org/keycloak/config/BootstrapAdminOptions.java

```diff
@@ -5,17 +5,23 @@ public class BootstrapAdminOptions {
     public static final String DEFAULT_TEMP_ADMIN_USERNAME = "temp-admin";
     public static final String DEFAULT_TEMP_ADMIN_SERVICE = DEFAULT_TEMP_ADMIN_USERNAME;
     public static final int DEFAULT_TEMP_ADMIN_EXPIRATION = 120;
+    public static final boolean DEFAULT_TEMP_ADMIN_IS_TEMPORARY = true;
     private static final String USED_ONLY_WHEN = " Used only when the master realm is created.";
     private static final String NON_CLI = " Use a non-CLI configuration option for this option if possible.";
 
+    // Since the Configuration object is not available in the WelcomeResource, we'll just pass it through a static
+    // handler. It's definitely a suboptimal approach but the alternatives are to either put it in the Profile
+    // (which doesn't feel right) or RealmModel (which seems like too much).
+    private static boolean isTemporaryRuntimeAdmin = true;
```

Comment:
@shawkins @vmuzikar Unfortunately I didn't find any better way to read the `--is-temporary` bootstrap parameter value from the `WelcomeResource`. This approach is not pretty but seems to work correctly. I'm open for suggestions on how to improve it (or we can proceed as is if it's OK).

Reply:
Is it truly necessary for the is-temporary option to influnce the WelcomeResource?

The alternative to this static variable to map the is-temporary option to an spi key and use Config.scope

We mused about providing the root configuration to Config via something like https://github.com/keycloak/keycloak/pull/39123 but it didn't move forward.

Reply:
> The alternative to this static variable to map the is-temporary option to an spi key and use Config.scope

Yes, that would be the correct approach.